### PR TITLE
Add radius foundation and tokens to storybook

### PR DESCRIPTION
**Pull request by jameskoster**
## What?
Adds radius foundations and tokens to storybook.

## Why?

- Documenting the foundations creates an important point of reference for updating (or creating new) components and UIs.
- Documenting the tokens makes them more discoverable.

## Testing Instructions
1. `npm run storybook`
2. Check out the new pages; 'Foundations / Design Language / Radius' and 'Tokens / Radius'

| Foundations page | Tokens page |
| --- | --- |
| Screenshot 2024-10-17 at 18 30 56 | Screenshot 2024-10-17 at 18 31 05 |

### CODE REVIEW

**Comment by jasmussen** [comment]
I think there's a question about the visual style we want to apply across these docs, it may be good to decide on a specific illustration style (which could be what you suggested, but it could also be [this](https://www.figma.com/design/bLa1mQsYSEIg1s0WzrigTi/Abstracted-Block-Editor-Illustrations?node-id=0-1&t=q1MC54DRhsx1SZb4-1)), and then roll it out consistently. 

I think as a first draft this is fine. What do you think? Nice work.

**Comment by ciampo** [comment]
Rebased on top of latest changes, and pushed a couple of improvements / fixes.

In particular, I refactored the table so that it can be a more generic and work for both radius and elevation tokens. It's still a bit hacky, and we will definitely want to refine it (alongside other docs componentry)

File: storybook/stories/foundations/design-language/radius.mdx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+		<td width="50%" valign="top">
+			<img src={ radiusDo } alt="Radius do" width="100%" />* Scale
+			application of radius with element or container size. * Use
+			`radius-round` for circles and `radius-full` for pills.
+		</td>
+		<td width="50%" valign="top">
+			<img src={ radiusDont } alt="Radius don't" width="100%" />* Don't
+			nest larger radii inside smaller radii. * Don't apply the same
+			radius value to container and immediate descendent.
+		</td>
```
Comment (tyxla) [inline storybook/stories/foundations/design-language/radius.mdx]: Let's use actual bullets here, the will currently appear as plain text:

```suggestion
		<td width="50%" valign="top">
			<img src={ radiusDo } alt="Radius do" width="100%" />
			* Scale application of radius with element or container size.
			* Use `radius-round` for circles and `radius-full` for pills.
		</td>
		<td width="50%" valign="top">
			<img src={ radiusDont } alt="Radius don't" width="100%" />
			* Don't nest larger radii inside smaller radii.
			* Don't apply the same radius value to container and immediate descendent.
		</td>
```

Before:
Screenshot 2024-10-22 at 18 11 11


After:

Screenshot 2024-10-22 at 18 10 58

File: storybook/stories/tokens/components.tsx [hunk 2]
```diff
@@ -36,12 +36,12 @@ export const ElevationTable = ( { tokens } ) => {
 						</td>
 						<td style={ { padding: '24px' } }>
 							<div
-								aria-label={ `An square showing an example of the '${ name }' elevation styles` }
+								aria-label={ `A square showing an example of the '${ name }' ${ tokenCategory } token` }
```
Comment (tyxla) [inline storybook/stories/tokens/components.tsx]: Won't really be a square for round/full, no? Should we use a more generic word like "shape"?

**Review by tyxla — APPROVED** [review]
This is great work @jameskoster! 🚀 

Left a few suggestions, but this is good to go anyway!

File: storybook/stories/tokens/components.tsx (hunk 2 above)
Comment (ciampo) [reply storybook/stories/tokens/components.tsx]: Sounds good

File: storybook/stories/foundations/design-language/radius.mdx (hunk 1 above)
Comment (ciampo) [reply storybook/stories/foundations/design-language/radius.mdx]: Well spotted — I believe it's my auto-formatting settings's fault. Adding an extra white line in between each list item correctly tells the markdown formatted that those are separate paragraphs.

Pushed a commit to fix it

**Comment by ciampo** [comment]
@tyxla Since I was the last one to work on this PR (and I'm not sure about Jay's availability this week), I've addressed the remaining feedback.

Will merge as soon as CI checks pass ✅

**Comment by tyxla** [comment]
Thanks @ciampo and @jameskoster 🙌

**Pull request merged**